lib/screens/settings/settings_screen.dart

```diff
@@ -56,6 +52,33 @@ class SettingsScreen extends StatelessWidget {
     );
   }
 
+  void _popdialog(BuildContext theContext){
+    showDialog(
+      context: theContext,
+      builder:(theContext) {
+        return AlertDialog(
+          title : Text('Log Out!'),
+          content: Text('Are you sure you want to logout?'),
+          actions: <Widget>[
+            FlatButton(
+              child: Text('Confirm'),
+              onPressed: () {
+                BlocProvider.of<AuthBloc>(theContext).add(JustLoggedOut());
+
+                Navigator.of(theContext).pop();
+                Navigator.of(theContext).pop();
```

Comment:
Why are you doing this (and why twice)? Bloc in `main.dart` should automatically handle logout event and show `LoginScreen`.
EDIT: Is it for hiding the two dialogs?

Reply:
Actually when I use only one pop(), only the dialog box closes and I'm still on the settings page. But when I use 2 it directs me to the login page.